### DISCUSSION TITLE
How do I read Operating Mode?

### DISCUSSION THREAD

**Discussion by Rustimation** [philippoo66/optolink-splitter#42] [Q&A] [answered]
The Viessmann API shows the Operating Mode (Betriebsart) with this filter: _'heating.circuits.1.operating.modes.active_'
I haven't been able to find the equivalent in my optoink parameter file (DP_VScotHO1_200_11.txt). It doesn't seem to be
_Aktuelle Betriebsart M2 (709) [HK_AktuelleBetriebsartM2~0x3500 (Byte)]_ as the result is alwys "2", regardless of the operating mode I set.
Does anybody know where to find it?

**Comment by philippoo66** [comment, marked as answer]
try 0x2323 (Betriebsart A1M1 [BedienteilBA]). the 350x, 450x do only apply if you have more than one circuit.

values are 
0: "Abschaltbetrieb"
1: "Nur Warmwasser"
2: "Heizen und WW"
3: "Dauernd Reduziert"
4: "Dauernd Normal"

don't know if 3 and 4 exist with DP_VScotHO1_200

**Comment by Rustimation** [comment]
Perfetto! I was looking in the wrong section. Interpreting the parameter file is more ike reverse polish notation i.e. confusing and complicated... But I'll manage. Thanks for the input.
Chris

**Comment by CM000n** [comment]
I'm interested in that too. Unfortunately, with the information under 0x2323 I have no information about which operating mode is actually present. For example, in the "Heating and hot water" operating mode, I can currently be in economy mode or normal mode. Depending on what is programmed via the schedule.
Is it really only possible to find out the current mode via the address for the schedule and the values returned there? Or is there still an address that can be used to find out the "actual" current operating mode and/or the "actual" current setpoint temperature for the "Heating and hot water" operating mode with schedule? 
(In my case I'm looking for a address for the Vitocrossal300)

**Reply by philippoo66** [reply]
Hi Simon!

With Vitocrossal it is probably a little bit different since the crossal do not have a valve 'inside'. 

Looking at DP_VScotHO1_200_30.txt (which should apply to Vitocrossal - did you check F8/F0?) I find
```
	Zeile  19:     - Aktuelle Betriebsart A1M1 (708) [HK_AktuelleBetriebsartA1M1~0x2500 (Byte)]
	Zeile  20:     - Externe Betriebsarten-umschaltung A1M1 (270) [ExtBetriebsartenumschaltung_A1M1~0x2549 (Byte)] HIDDEN:(ZuordExtBetrUmsch="0 inaktiv" OR ZuordExtBetrUmsch="2 M2" OR ZuordExtBetrUmsch="4 M3" OR ZuordExtBetrUmsch="6 M2 + M3")
	Zeile  36:     - Aktuelle Betriebsart M2 (709) [HK_AktuelleBetriebsartM2~0x3500 (Byte)]
	Zeile  37:     - Externe Betriebsarten-umschaltung M2 (271) [ExtBetriebsartenumschaltung_M2~0x3549 (Byte)] HIDDEN:(ZuordExtBetrUmsch="0 inaktiv" OR ZuordExtBetrUmsch="1 A1/M1" OR ZuordExtBetrUmsch="4 M3" OR ZuordExtBetrUmsch="5 A1/M1 + M3")
	Zeile  53:     - Aktuelle Betriebsart M3 (710) [HK_AktuelleBetriebsartM3~0x4500 (Byte)]
	Zeile  54:     - Externe Betriebsarten-umschaltung M3 (272) [ExtBetriebsartenumschaltung_M3~0x4549 (Byte)] HIDDEN:(ZuordExtBetrUmsch="0 inaktiv" OR ZuordExtBetrUmsch="1 A1/M1" OR ZuordExtBetrUmsch="2 M2" OR ZuordExtBetrUmsch="3 A1/M1 + M2")
	Zeile 102:     - Betriebsart A1M1 (92) [BedienteilBA_GWGA1~0x2323 (Byte)]
	Zeile 120:     - Betriebsart M2 (94) [BedienteilBA_GWGM2~0x3323 (Byte)]
	Zeile 138:     - Betriebsart M3 (7219) [BedienteilBA_GWGM3~4323 (Byte)]
	Zeile 179:     - (D5) Wirkung Betriebsarten-umschaltung A1M1 (2880) [KD5_KonfiWirkungExtBA_UmschaltungA1M1~0x27D5 (Byte)] HIDDEN:(ZuordExtBetrUmsch="0 inaktiv" OR ZuordExtBetrUmsch="2 M2" OR ZuordExtBetrUmsch="4 M3" OR ZuordExtBetrUmsch="6 M2 + M3")
	Zeile 183:     - (F2) Zeitbegrenzung Party/ Betriebsarten-Umschaltung A1 (2962) [KF2_KonfiPartyzeitGWGA1~0x27F2 (Byte)]
	Zeile 199:     - (D5) Wirkung Betriebsarten-umschaltung M2 (2881) [KD5_KonfiWirkungExtBA_UmschaltungM2~0x37D5 (Byte)] HIDDEN:(ZuordExtBetrUmsch="0 inaktiv" OR ZuordExtBetrUmsch="1 A1/M1" OR ZuordExtBetrUmsch="4 M3" OR ZuordExtBetrUmsch="5 A1/M1 + M3")
	Zeile 203:     - (F2) Zeitbegrenzung Party/ Betriebsarten-Umschaltung M2 (2963) [KF2_KonfiPartyzeitGWGM2~0x37F2 (Byte)]
	Zeile 218:     - (D5) Wirkung Betriebsarten-umschaltung M3 (2882) [KD5_KonfiWirkungExtBA_UmschaltungM3~0x47D5 (Byte)] HIDDEN:(ZuordExtBetrUmsch="0 inaktiv" OR ZuordExtBetrUmsch="1 A1/M1" OR ZuordExtBetrUmsch="2 M2" OR ZuordExtBetrUmsch="3 A1/M1 + M2")
	Zeile 321:     - (D5) Wirkung Betriebsarten-umschaltung A1M1 (2880) [KD5_KonfiWirkungExtBA_UmschaltungA1M1~0x27D5 (Byte)] HIDDEN:(ZuordExtBetrUmsch="0 inaktiv" OR ZuordExtBetrUmsch="2 M2" OR ZuordExtBetrUmsch="4 M3" OR ZuordExtBetrUmsch="6 M2 + M3")
	Zeile 328:     - (F2) Zeitbegrenzung Party/ Betriebsarten-Umschaltung A1 (2962) [KF2_KonfiPartyzeitGWGA1~0x27F2 (Byte)]
	Zeile 350:     - (D5) Wirkung Betriebsarten-umschaltung M2 (2881) [KD5_KonfiWirkungExtBA_UmschaltungM2~0x37D5 (Byte)] HIDDEN:(ZuordExtBetrUmsch="0 inaktiv" OR ZuordExtBetrUmsch="1 A1/M1" OR ZuordExtBetrUmsch="4 M3" OR ZuordExtBetrUmsch="5 A1/M1 + M3")
	Zeile 357:     - (F2) Zeitbegrenzung Party/ Betriebsarten-Umschaltung M2 (2963) [KF2_KonfiPartyzeitGWGM2~0x37F2 (Byte)]
	Zeile 502:     - Aktuelle Betriebsart A1M1 (708) [HK_AktuelleBetriebsartA1M1~0x2500 (Byte)]
	Zeile 503:     - Externe Betriebsarten-umschaltung A1M1 (270) [ExtBetriebsartenumschaltung_A1M1~0x2549 (Byte)] HIDDEN:(ZuordExtBetrUmsch="0 inaktiv" OR ZuordExtBetrUmsch="2 M2" OR ZuordExtBetrUmsch="4 M3" OR ZuordExtBetrUmsch="6 M2 + M3")
	Zeile 522:     - Aktuelle Betriebsart M2 (709) [HK_AktuelleBetriebsartM2~0x3500 (Byte)]
	Zeile 523:     - Externe Betriebsarten-umschaltung M2 (271) [ExtBetriebsartenumschaltung_M2~0x3549 (Byte)] HIDDEN:(ZuordExtBetrUmsch="0 inaktiv" OR ZuordExtBetrUmsch="1 A1/M1" OR ZuordExtBetrUmsch="4 M3" OR ZuordExtBetrUmsch="5 A1/M1 + M3")
	Zeile 541:     - Aktuelle Betriebsart M3 (710) [HK_AktuelleBetriebsartM3~0x4500 (Byte)]
	Zeile 542:     - Externe Betriebsarten-umschaltung M3 (272) [ExtBetriebsartenumschaltung_M3~0x4549 (Byte)] HIDDEN:(ZuordExtBetrUmsch="0 inaktiv" OR ZuordExtBetrUmsch="1 A1/M1" OR ZuordExtBetrUmsch="2 M2" OR ZuordExtBetrUmsch="3 A1/M1 + M2")
	Zeile 574:     - ext. Anfordern/ Betriebsarten-umschaltung (20) [AE_Eingang1~0x0A80 (Byte)]
```

You need to figure out which way your WW heating is controlled an then pick the according datapoint...

**Reply by CM000n** [reply]
Thanks @philippoo66 , but I have already checked these addresses.

- Under 0x3500 / 0x4500 I only get back 2 as a value.
- Under 0x3323 / 0x4323 I get the values 1,2,3,4 (In my case, it is usually set to 1 “Hot water only” or 2 “Hot water and heating”)
- Under 0x3549 / 0x4549 & 0x37D5 / 0x47D5 I only get back 0 as value.

So yes, no information about whether a reduced temperature is currently present or the normal value. 😞 

I get the best information under the addresses 0x3544 / 0x4544. They can be used to get the current setpoint temperature.
However, it does not return the current setpoint room temperature that I am looking for (21°C normal / 18°C reduced), but the current setpoint temperature of the flow.

**Comment by FransOv** [comment]
I found operating mode at 0xB000 (HK1) and 0xB001 (HK2) for the 300-G. 
5 – dauernd normal,4 – dauernd reduziert,2 – Heizen/Kühlen/WW,1 – Nur WW,0 – Abschaltbetrieb,7 – nur Kühlen,6 – normal Abschalt
